themis 1.1.0 blog post (#376)
posit-dev/open-source-website: content/blog/themis-1-1-0/index.md

- [X] Open a PR against main, then comment /deploy-preview (this is a fork)
-->

I'm very happy to announce that [themis 1.1.0](https://themis.tidymodels.org/) is now on CRAN.
themis provides extra recipes steps for dealing with unbalanced data.
You can install it with:

``` r
install.packages("themis")
```

This release provides a substantial amount of new features.
We will cover the highlights in this blog post, which include:
11 new steps, setting sampling targets per class, and more distance metrics.
See the [news file](https://themis.tidymodels.org/news/index.html#themis-110) for a complete list of changes in this release.

To get started,
load the tidymodels and themis packages:

``` r
library(tidymodels)
library(themis)

set.seed(1234)
```

## New sampling steps

This release adds support for 11 new steps for under- and over-sampling of data.
Making for a more complete picture of the commonly discussed methods in this category.
Each step also ships a direct-implementation counterpart (`enn()`, `smogn()`, and so on) for use outside a recipe.

The new steps fall into four groups.

**Cleaning-based under-sampling** removes observations that sit in the wrong neighborhood:

> - `step_enn()` applies the Edited Nearest Neighbors rule.

Reviewer: ```suggestion
- `step_enn()` applies the [Edited Nearest Neighbors rule]("Asymptotic Properties of Nearest Neighbor Rules Using Edited Data").
```

> - `step_ncl()` applies the Neighborhood Cleaning Rule.

Reviewer: ```suggestion
- `step_ncl()` applies the [Neighborhood Cleaning Rule](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22Improving+identification+of+difficult+small+classes+by+balancing+class+distribution%22&btnG=).
```

> - `step_cnn()` uses Condensed Nearest Neighbors.

Reviewer: ```suggestion
- `step_cnn()` uses [Condensed Nearest Neighbors](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22The+condensed+nearest+neighbor+rule%22&btnG=).
```

> - `step_oss()` uses One-Sided Selection.

Reviewer: ```suggestion
- `step_oss()` uses [One-Sided Selection](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22An+Effective+Over-sampling+Method+for+Imbalanced+Data+Sets+Classification%22&btnG=).
```

**Selection-based under-sampling** picks a smaller set of representatives instead:

> - `step_cluster_centroids()` replaces each class with one representative per k-means cluster.

Reviewer: ```suggestion
- `step_cluster_centroids()` replaces each class with [one representative per k-means cluster](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22Clustering-based+undersampling+in+class-imbalanced+data%22&btnG=).
```

> - `step_instance_hardness()` removes the observations that are hardest to classify.

Reviewer: ```suggestion
- `step_instance_hardness()` removes the [observations that are hardest to classify](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22An+instance+level+analysis+of+data+complexity%22&btnG=).
```

**Over-sampling** gains three new variants:

> - `step_kmeans_smote()` generates new examples only inside clusters where the minority class dominates.

Reviewer: ```suggestion
- `step_kmeans_smote()` generates new examples only inside [clusters where the minority class dominates](https://scholar.google.com/scholaras_sdt=0%2C7&q=%22Improving+imbalanced+learning+through+a+heuristic+oversampling+method+based+on+k-means+and+SMOTE%22&btnG=).
```

> - `step_svmsmote()` concentrates them near the decision boundary found by a support vector machine.

Reviewer: ```suggestion
- `step_svmsmote()` concentrates them near the [ecision boundary found by a support vector machine](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22Borderline+over-sampling+for+imbalanced+data+classification%22&btnG=).
```

> - `step_smoten()` handles data where every predictor is categorical.

Reviewer: ```suggestion
- `step_smoten()` handles data where [every predictor is categorical](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22Smote%3A+Synthetic+minority+over-sampling+technique%22&btnG=.
```

Lastly we have a step that applies the SMOTE idea to regression:

> - `step_smogn()` over-samples rare regions of a numeric outcome and under-samples the common ones.

Reviewer: ```suggestion
- `step_smogn()` [over-samples rare regions of a numeric outcome and under-samples the common ones](https://scholar.google.com/scholar?&as_sdt=0%2C7&q=%22SMOGN%3A+a+pre-processing+approach+for+imbalanced+regression%22&btnG=.
```
